LFS171x/docs/introduction-to-hyperledger-fabric.md

```diff
@@ -1 +1,1547 @@
+
+There are two ways to develop smart contracts with Hyperledger Fabric:
+
+ - Code individual contracts into standalone instances of chaincode
+ - (More efficient way) Use chaincode to create decentralized applications that manage the lifecycle of one or multiple types of business contracts, and let the end users instantiate instances of contracts within these applications.
+ 
+### Chaincode Key APIs
+
+An important interface that you can use when writing your chaincode is defined by Hyperledger Fabric - <span style="color:cyan">_ChaincodeStub_</span> and <span style="color:cyan">_ChaincodeStubInterface_</span>. The ChaincodeStub provides functions that allow you to interact with the underlying ledger to query, update, and delete assets. The key APIs for chaincode include:
+
+ - **func (stub \*<span style="color:cyan">ChaincodeStub</span>) GetState(key <span style="color:cyan">string</span>) ([]<span style="color:cyan">byte, error</span>)**
+ <br />Returns the value of the specified _key_ from the ledger. Note that **GetState** doesn't read data from the Write set, which has not been committed to the ledger. In other words, **GetState** doesn't consider data modified by **PutState** that has not been committed. If the key does not exist in the state database, 
+ **\(nil, nil)** is returned.
+ - **func (stub \*<span style="color:cyan">ChaincodeStub</span>) PutState(key <span style="color:cyan">string</span>, value []<span style="color:cyan">byte</span>) <span style="color:cyan">error</span>**
+ <br />Puts the specified _key_ and _value_ into the transaction's Write set as a data-write proposal. **PutState** doesn't affect the ledger until the transaction is validated and successfully committed.
+ - **func (stub \*<span style="color:cyan">ChaincodeStub)</span> DelState(key <span style="color:cyan">string</span>) <span style="color:cyan">error</span>**
+ <br />Records the specified key to be deleted in the Write set of the transaction proposal. The _key_ and its _value_ will be deleted from the ledger when the transaction is validated and successfully committed.
+
+### Overview of a Chaincode Program
+
+When creating a chaincode, there are two methods that you will need to implement:
+
+ - **Init**
+ <br />Called when a chaincode receives an **_instantiate_** or **_upgrade_** transaction. This is where you will initialize any application state.
+ - **Invoke**
+ <br />Called when the **_invoke_** transaction is received to process any transaction proposals.
+
+As a developer, you must create both an **Init** and an **Invoke** method within your chaincode. The chaincode must be installed using the **peer chaincode install** command, and instantiated using the **peer chaincode instantiate** command before the chaincode can be invoked. Then, transactions can be created using the **peer chaincode invoke** or peer **chaincode query commands**.
```

Comment:
"peer chaincode install", "peer chaincode instantiate", and "peer chaincode invoke" should all be bold code.